xds/internal/xdslbregistry/converter.go

```diff
@@ -0,0 +1,288 @@
+		if cont {
+			continue
+		}
+		return json, err
+	}
+	return nil, fmt.Errorf("no supported policy found in policy list +%v", lbPolicy)
+}
+
+type ringHashConverter struct{}
+
+func (rhc *ringHashConverter) convertToServiceConfig(rawProto []byte, depth int) (json.RawMessage, bool, error) {
+	if !envconfig.XDSRingHash { // either have this as part of converter or top level - not a switch so top level
+		return nil, true, nil
+	}
+	rhProto := &v3ringhashpb.RingHash{}
+	if err := proto.Unmarshal(rawProto, rhProto); err != nil {
+		return nil, false, fmt.Errorf("failed to unmarshal resource: %v", err)
+	}
+	if rhProto.GetHashFunction() != v3ringhashpb.RingHash_XX_HASH {
+		return nil, false, fmt.Errorf("unsupported ring_hash hash function %v", rhProto.GetHashFunction())
+	}
+
+	var minSize, maxSize uint64 = defaultRingHashMinSize, defaultRingHashMaxSize
+	if min := rhProto.GetMinimumRingSize(); min != nil {
+		minSize = min.GetValue()
+	}
+	if max := rhProto.GetMaximumRingSize(); max != nil {
+		maxSize = max.GetValue()
+	}
+
+	rhCfg := &ringhash.LBConfig{
+		MinRingSize: minSize,
+		MaxRingSize: maxSize,
+	}
+
+	rhCfgJSON, err := json.Marshal(rhCfg)
+	if err != nil {
+		return nil, false, fmt.Errorf("error marshaling JSON for type %T: %v", rhCfg, err)
+	}
+	return makeBalancerConfigJSON(ringhash.Name, rhCfgJSON), false, nil
+}
+
+type roundRobinConverter struct{}
+
+func (rrc *roundRobinConverter) convertToServiceConfig([]byte, int) (json.RawMessage, bool, error) {
+	return makeBalancerConfigJSON("round_robin", json.RawMessage("{}")), false, nil
+}
+
+type wrrLocalityConverter struct {
+	depth int
+}
+
+type wrrLocalityLBConfig struct {
+	ChildPolicy json.RawMessage `json:"childPolicy,omitempty"`
+}
+
+func (wlc *wrrLocalityConverter) convertToServiceConfig(rawProto []byte, depth int) (json.RawMessage, bool, error) {
+	wrrlProto := &v3wrrlocalitypb.WrrLocality{}
+	if err := proto.Unmarshal(rawProto, wrrlProto); err != nil {
+		return nil, false, fmt.Errorf("failed to unmarshal resource: %v", err)
+	}
+	epJSON, err := convertToServiceConfig(wrrlProto.GetEndpointPickingPolicy(), depth+1)
+	if err != nil {
+		return nil, false, fmt.Errorf("error converting endpoint picking policy: %v for %+v", err, wrrlProto)
+	}
+	wrrLCfg := wrrLocalityLBConfig{
+		ChildPolicy: epJSON,
+	}
+
+	lbCfgJSON, err := json.Marshal(wrrLCfg)
+	if err != nil {
+		return nil, false, fmt.Errorf("error marshaling JSON for type %T: %v", wrrLCfg, err)
+	}
+	return makeBalancerConfigJSON(wrrlocality.Name, lbCfgJSON), false, nil
+}
+
+type weightedRoundRobinConverter struct{}
+
+func (wrrc *weightedRoundRobinConverter) convertToServiceConfig(rawProto []byte, depth int) (json.RawMessage, bool, error) {
+	cswrrProto := &v3clientsideweightedroundrobinpb.ClientSideWeightedRoundRobin{}
+	if err := proto.Unmarshal(rawProto, cswrrProto); err != nil {
+		return nil, false, fmt.Errorf("failed to unmarshal resource: %v", err)
+	}
+	wrrLBConfig := &wrrLBConfig{}
```

Comment:
Nit: `Cfg`?  1. doesn't shadow, 2. matches pattern established.

Reply:
Yeah, good point on 2. Is shadowing types bad? I've def had a lot of type := type declarations.

Reply:
Sg.